packages/react-reconciler/src/ReactFiberThrow.js

```diff
@@ -175,6 +175,35 @@ function attachPingListener(
   }
 }
 
+const avoidedThenable = {then() {}};
+
+export function unstable_avoidThisRenderImpl(
+  root: FiberRoot,
+  returnFiber: Fiber,
+  sourceFiber: Fiber,
+  value: Thenable,
+  renderExpirationTime: ExpirationTime,
+) {
+  let hasInvisibleParentBoundary = hasSuspenseContext(
+    suspenseStackCursor.current,
+    (InvisibleParentSuspenseContext: SuspenseContext),
+  );
+  if (
+    hasInvisibleParentBoundary
+    // fallback showing
+  ) {
+    throwException(
+      root,
+      returnFiber,
+      sourceFiber,
+      avoidedThenable,
+      renderExpirationTime,
+    );
+  } else {
+    throw value;
```

Comment:
I think this part is wrong.

If we're inside already visible content, then we should continue rendering without triggering the parent fallback *and* delay commit.

So this should call `renderDidSuspendDelayIfPossible()`, which will change the root exit status to RootDidSuspendWithDelay. And it should attach a ping listener, so we can unblock the commit.

In the next PR, when we implement the ref counting strategy, this will instead change the root exit status to a new value (i.e. RootDidSuspendWithDelayRefCounting, but we'll think of a better name) and attach a special kind of ping listener that decrements the ref count.

Reply:
Need to make sure you have a test that covers this branch, btw. The test should mount a Suspense boundary, then in a `useTransition` update, mount a new image. The image will delay the commit instead of hiding the already-visible content.